apps/hash-api/src/graph/ensure-system-graph-is-initialized/migrate-ontology-types/migrations/009-create-office-file-types.migration.ts

```diff
@@ -0,0 +1,168 @@
+      migrationState,
+      instantiator: anyUserInstantiator,
+    },
+  );
+
+  const _pdfFileEntityType = await createSystemEntityTypeIfNotExists(
+    context,
+    authentication,
+    {
+      entityTypeDefinition: {
+        allOf: [documentFileEntityType.schema.$id],
+        title: "PDF Document",
+        description: "A PDF document.",
+      },
+      webShortname: "hash",
+      migrationState,
+      instantiator: anyUserInstantiator,
+    },
+  );
+
+  const _docxFileEntityType = await createSystemEntityTypeIfNotExists(
+    context,
+    authentication,
+    {
+      entityTypeDefinition: {
+        allOf: [documentFileEntityType.schema.$id],
+        title: "DOCX Document",
+        description: "A Microsoft Word document.",
+      },
+      webShortname: "hash",
+      migrationState,
+      instantiator: anyUserInstantiator,
+    },
+  );
+
+  /**
+   * Step 2. Create the `Slideshow` entity type and its child entity types.
+   */
+
+  const slideshowFileEntityType = await createSystemEntityTypeIfNotExists(
+    context,
+    authentication,
+    {
+      entityTypeDefinition: {
+        allOf: [fileEntityTypeId],
+        title: "Presentation File",
+        description: "A presentation file.",
+        properties: [
+          {
+            propertyType:
+              "https://blockprotocol.org/@blockprotocol/types/property-type/textual-content/v/2",
+            required: false,
+          },
+        ],
+      },
+      webShortname: "hash",
+      migrationState,
+      instantiator: anyUserInstantiator,
+    },
+  );
+
+  const _pptxFileEntityType = await createSystemEntityTypeIfNotExists(
+    context,
+    authentication,
+    {
+      entityTypeDefinition: {
+        allOf: [slideshowFileEntityType.schema.$id],
+        title: "PPTX Presentation",
+        description: "A Microsoft PowerPoint presentation.",
+      },
+      webShortname: "hash",
+      migrationState,
+      instantiator: anyUserInstantiator,
+    },
+  );
+
+  /**
+   * Step 3. Create the `Spreadsheet` entity type and its child entity types.
+   */
+
+  const spreadsheetFileEntityType = await createSystemEntityTypeIfNotExists(
+    context,
+    authentication,
+    {
+      entityTypeDefinition: {
+        allOf: [fileEntityTypeId],
+        title: "Spreadsheet File",
+        description: "A spreadsheet file.",
+        /** @todo: add property definition */
+      },
+      webShortname: "hash",
+      migrationState,
+      instantiator: anyUserInstantiator,
+    },
+  );
+
+  const _microsoftExcelFileEntityType = await createSystemEntityTypeIfNotExists(
+    context,
+    authentication,
+    {
+      entityTypeDefinition: {
+        allOf: [spreadsheetFileEntityType.schema.$id],
+        title: "XLSX Spreadsheet",
+        description: "A Microsoft Excel spreadsheet.",
+      },
+      webShortname: "hash",
+      migrationState,
+      instantiator: anyUserInstantiator,
+    },
+  );
+
+  return migrationState;
+};
```

Comment:
@nonparibus should I omit the "Spreadsheet" and "XLSX Spreadsheet" definitions in this PR, given we are deferring defining its properties/links?